pom.xml

```diff
@@ -383,6 +383,11 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>org.checkerframework</groupId>
+      <artifactId>checker-qual</artifactId>
+      <version>3.12.0</version>
```

Comment:
@Vyom-Yadav why aren't we using newest version?

Reply:
https://checkerframework.org/manual/#installation

3.25

Reply:
Hmmm, why we are not running our shell script for version check on each PR to prevent such problem?
Is there any objections?

Reply:
new CI idea is moved to https://github.com/checkstyle/checkstyle/issues/12193

Reply:
We aren't using the latest version because of 
```cmnd
[WARNING] 
Dependency convergence error for org.checkerframework:checker-qual:jar:3.12.0:compile paths to dependency are:
+-org.hibernate.search:hibernate-search-build-config:jar:6.2.0-SNAPSHOT
  +-com.puppycrawl.tools:checkstyle:jar:10.3.4-SNAPSHOT:compile
    +-com.google.guava:guava:jar:31.1-jre:compile
      +-org.checkerframework:checker-qual:jar:3.12.0:compile
and
+-org.hibernate.search:hibernate-search-build-config:jar:6.2.0-SNAPSHOT
  +-com.puppycrawl.tools:checkstyle:jar:10.3.4-SNAPSHOT:compile
    +-org.checkerframework:checker-qual:jar:3.25.0:compile

[ERROR] Rule 0: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
Failed while enforcing releasability. See above detailed error message.
```

Reply:
such problems should not block us to use latest version.
We can make version of checker-qual as variable and set it to 3.12.0 before execution of hibernate.
Or we can ask hibernate project to make variable on their side to let us not conflict (we will do this after we become confident that we are going to use checker for sure).

Reply:
Problem will occur in our codebase too if we use 3.25.0 without exclusions, I am closing this in favor of #12194, we can continue the discussion over there.